monai/metrics/seg_metric_utils.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from monai.networks import one_hot
+from monai.utils import MetricReduction
+
+
```

Comment:
I think these `do_onehot`, `do_actication`, and `do_binarization` APIs are duplicated with `Activations` and `AsDiscrete` post transforms.
@wyli , I think maybe we can drop these logic in all the metrics directly to make them more clear? If user is writing PyTorch program with our metrics, they can call the post transforms in validation loop before calling metrics, if writing program based on our workflows, it already contains the `post transforms`.

Thanks.